resources/apiserver-proxy/templates/deployment.yaml

```diff
@@ -53,9 +86,9 @@ spec:
         - --v=9
         - --oidc-issuer=https://dex.{{ .Values.global.ingress.domainName }}
         - --oidc-clientID=kyma-client
-        {{ if not .Values.global.environment.gardener }}
-        - --oidc-ca-file=/etc/dex-tls-cert/tls.crt
-        {{ end }}
+      {{- if or (eq .Values.global.certificates.type "legacy") (eq .Values.global.certificates.type "xip") }}
```

Comment:
I do not like it. This component should not now anything about certificates component. Provide a meaningful flag that is self-explanatory. In this case it is selfSigned or something like this. But I would remove the condition completely. You can always pass kyma tls certificate. It should work either for proper trusted certificates and self-signed.